File: leetcode/20.Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) % 2 != 0:
            return False

        stack = []

        for char in s:
            if char in ['(', '[', '{']:
                stack.append(char)
            else:
                if char == ')':
                    if not stack or stack.pop() != '(':
                        return False
                elif char == ']':
                    if not stack or stack.pop() != '[':
                        return False
                elif char == '}':
                    if not stack or stack.pop() != '{':
                        return False
        if len(stack) > 0:
            return False
        return True

File: leetcode/20.Valid Parentheses.ts
function isValid(s: string): boolean {
	if (s.length % 2 !== 0) return false;

	let stack: string[] = [];
	const len = s.length;

	for (let i = 0; i < len; i += 1) {
		if (s[i] === '(' || s[i] === '[' || s[i] === '{') stack.push(s[i]);
		else {
			// s[i]가 } ) ] 인 경우
			const pop = stack.pop();
			if (s[i] === '}' && pop === '{') continue;
			else if (s[i] === ']' && pop === '[') continue;
			else if (s[i] === ')' && pop === '(') continue;
			else return false;
		}
	}
	if (stack.length !== 0) return false;
	return true;
}
// console.log('🚀 ~ answer:', isValid('([)]'));
// console.log('🚀 ~ answer:', isValid('((([[[{{{}}}]]])))'));
// console.log('🚀 ~ answer:', isValid('()[]{}'));
// console.log('🚀 ~ answer:', isValid('([}}])'));
// console.log('🚀 ~ answer:', isValid('[({])}'));
